azurerm/internal/services/eventgrid/data_source_eventgrid_topic.go

```diff
@@ -0,0 +1,93 @@
+package eventgrid
+
+import (
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-sdk/helper/schema"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/helpers/azure"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/clients"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/tags"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/internal/timeouts"
+	"github.com/terraform-providers/terraform-provider-azurerm/azurerm/utils"
+)
+
+func dataSourceArmEventGridTopic() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceArmEventGridTopicRead,
+
+		Timeouts: &schema.ResourceTimeout{
+			Read: schema.DefaultTimeout(5 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
```

Comment:
Could we validate this?

Reply:
Added the **NoEmptyStrings** validation to this and the **resource_azurerm_eventgrid_topic** resource. Let me know if you had a different type of validation in mind.